crates/cli/src/main.rs

```diff
@@ -300,13 +300,19 @@ async fn start_program() -> Result<(), AnvilZksyncError> {
 
     // We start the node executor now so it can receive and handle commands
     // during replay. Otherwise, replay would send commands and hang.
-    tokio::spawn(async move {
-        if let Err(err) = node_executor.run().await {
-            let error = err.context("Node executor ended with error");
-            sh_err!("{:?}", error);
-
-            let _ = telemetry.track_error(Box::new(error.as_ref())).await;
-        }
+    std::thread::spawn(move || {
+        let rt = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()
+            .expect("Failed to build current-thread runtime");
+        rt.block_on(async move {
+            if let Err(err) = node_executor.run().await {
+                let error = err.context("Node executor ended with error");
+                sh_err!("{:?}", error);
+
+                let _ = telemetry.track_error(Box::new(error.as_ref())).await;
+            }
+        });
```

Comment:
This is required, at least to my understanding, as in `run_l2_txs` we use a pointer for `storage` which `vm::new` expects, otherwise I would be able to make use of `Arc<Mutex` instead. 

Please advise if there is better approach here or if my understanding above is incorrect.

Reply:
Indeed, I have spent the last couple of weeks trying to integrate `BatchExecutor` from zksync-era to solve this exact problem (among others) and to be honest with you it is not going that great (many small issues that I can expand upon if there is interest). I can *probably* reduce the scope to preserve the existing behaviour with having exactly one block per batch to at least unblock this specific problem.

Reply:
Alright here is the smaller version of my original work that focuses purely on `BatchExecutor` (hence making `run_tx` async) - https://github.com/matter-labs/anvil-zksync/pull/614

I believe we can close this PR